### ISSUE TITLE
Layout/RescueEnsureAlignment issue with nested begin

### ISSUE THREAD

**Issue by dreyks** [rubocop-hq/rubocop#6603]
Looks similar to #6433 

## Expected behavior

No error emitted

## Actual behavior

Cop expects `rescue` to be aligned with `begin`

## Steps to reproduce the problem

```ruby
def some_method
  @ivar ||= begin
     1
  rescue
    2
  end
end
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ [bundle exec] rubocop -V
0.61.1 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-darwin17)
```

**Comment by jaredbeck** [comment]
I believe this cop is working as intended. Here are some options you have:

```ruby
# A
v ||= begin
        1
      rescue
        2
      end

# B
def one_two
  1
rescue
  2
end

v ||= one_two

# C
v ||=
  begin
    1
  rescue
    2
  end
```

I prefer B, personally. Hope that helps!

**Comment by jaredbeck** [comment]
Tests improved by https://github.com/rubocop-hq/rubocop/pull/6606

**Comment by dreyks** [comment]
Hm... so the following code while being contrived and generally not good is perfectly fine with rubocop
```ruby
case something
when something_else
  whatever = case nested
  when no_matter
    ret
  end
end
```
the nested `when` is not required to be placed relative to outer indentations

why is rescue different then?

**Comment by bbatsov** [comment]
@dreyks Likely an oversight. 

Generally some alignment cops have configurable styles to allow one of the two common styles of alignment, but this is not one of them (yet). I guess someone can tackle this down the road.

**Comment by dreyks** [comment]
ok cool. I might look into it if I have some spare time

**Comment by oehlschl** [comment]
Here after upgrading from 0.58 to 0.61, as this cop triggered many new failures for previously allowed syntax:

```
result = begin
  # do something
rescue
  # do something else
end
```

Since this was previously valid, and since a similar format is allowed for Ruby 2.5 do/rescue blocks (per https://github.com/rubocop-hq/rubocop/pull/6437), I'll also put in a vote for either loosening or parameterizing this.

**Comment by oehlschl** [comment]
Also, FWIW, I'm guessing you can close this in favor of https://github.com/rubocop-hq/rubocop/issues/6254, which seems like one of the more popular threads for this issue/request.

**Comment by Drenmi** [comment]
Closing as duplicate of #6254. Thanks @oehlschl! 🙇
